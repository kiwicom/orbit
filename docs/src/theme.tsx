import { defaultTokens } from "@kiwicom/orbit-design-tokens";

export default {
  orbit: { ...defaultTokens, widthBreakpointDesktop: 1024 },
  transitions: true,
  rtl: false,
};
