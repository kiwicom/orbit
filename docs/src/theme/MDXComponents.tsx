// import React from "react";
// import MDXComponents from "@theme-original/MDXComponents";
// import {
//   Alert,
//   Heading,
//   Separator,
//   Text,
//   TextLink,
//   TableHead,
//   TableBody,
//   TableRow,
//   TableCell,
// } from "@kiwicom/orbit-components";
// import { NewWindow } from "@kiwicom/orbit-components/icons";
// import { SpaceAfter } from "@kiwicom/orbit-components/lib/common/types";
// import { Type } from "@kiwicom/orbit-components/lib/Heading/types";
// import Link from "@docusaurus/Link";

// import BlockQuote from "../../ignore/BlockQuote";
// import HeadingWithLink from "../../ignore/HeadingWithLink";
// import { InlineCode, CodeBlock } from "../../ignore/Code";
// import useIsUrlExternal from "../hooks/useIsUrlExternal";
// import PropsTable from "../../ignore/docs-table";
// // Import custom components to make them available globally
// import Palette from "../../ignore/Palette";
// import IconList from "../../ignore/IconList";
// import IllustrationList from "../../ignore/IllustrationList";
// import ElevationLevel from "../../ignore/ElevationLevel";
// import ImagesDownloadList from "../../ignore/ImagesDownloadList";
// import { ComponentDesignTokens, GlobalDesignTokens } from "../../ignore/DesignTokensList";
// import TailwindClassnames from "../../ignore/TailwindClassnames";
// import GithubContributors from "../../ignore/Github";
// import { ComponentStatusList } from "../../ignore/ComponentStatus";
// import FigmaIframe from "../../ignore/FigmaIframe";
// import InlineToken from "../../ignore/InlineToken";
// import DocLayout from "../../ignore/DocLayout";

// export const p = ({ children }: React.HTMLAttributes<HTMLParagraphElement>) => (
//   <Text>{children}</Text>
// );

// type HeadingTag = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";

// function createHeadingComponent(
//   tag: HeadingTag,
//   { type, spaceAfter }: { type: Type } & SpaceAfter,
// ) {
//   function HeadingComponent({ noId, children }: { noId?: boolean; children: React.ReactNode }) {
//     const level = Number(tag.slice(1)) - 2;

//     return (
//       <HeadingWithLink level={level} noId={noId} spaceAfter={spaceAfter}>
//         <Heading as={tag} type={type}>
//           {children}
//         </Heading>
//       </HeadingWithLink>
//     );
//   }

//   HeadingComponent.displayName = `TOC(${tag})`;
//   return HeadingComponent;
// }

// export const h1 = createHeadingComponent("h1", { type: "title0", spaceAfter: "small" });
// export const h2 = createHeadingComponent("h2", { type: "title2", spaceAfter: "small" });
// export const h3 = createHeadingComponent("h3", { type: "title3", spaceAfter: "normal" });
// export const h4 = createHeadingComponent("h4", { type: "title4", spaceAfter: "smallest" });
// export const h5 = createHeadingComponent("h5", { type: "title5" });
// export const h6 = createHeadingComponent("h6", { type: "title5" });

// export const hr = () => <Separator spaceAfter="largest" />;

// export const ol = ({ children }: React.OlHTMLAttributes<HTMLOListElement>) => (
//   <ol className="ml-400 [&_ol]:m-300 [&_ul]:m-300 list-outside list-decimal">{children}</ol>
// );

// export const blockquote = BlockQuote;

// export const table = ({ children }: React.TableHTMLAttributes<HTMLTableElement>) => (
//   <PropsTable>{children}</PropsTable>
// );

// export const thead = ({ children }: React.HTMLAttributes<HTMLTableSectionElement>) => (
//   <TableHead>{children}</TableHead>
// );

// export const tbody = ({ children }: React.HTMLAttributes<HTMLTableSectionElement>) => (
//   <TableBody>{children}</TableBody>
// );

// export const tr = ({ children }: React.HTMLAttributes<HTMLTableRowElement>) => (
//   <TableRow>{children}</TableRow>
// );

// export const td = ({ children, align, valign }: React.TdHTMLAttributes<HTMLTableCellElement>) => (
//   <TableCell
//     as="td"
//     align={align === "left" || align === "center" || align === "right" ? align : undefined}
//     whiteSpace="normal"
//     verticalAlign={valign}
//   >
//     {children}
//   </TableCell>
// );

// export const th = ({ children }: React.ThHTMLAttributes<HTMLTableCellElement>) => (
//   <TableCell as="th">{children}</TableCell>
// );

// export const ul = ({ children }: React.HTMLAttributes<HTMLUListElement>) => (
//   <ul className="ml-400 [&_ol]:m-300 [&_ul]:m-300 list-outside list-disc">{children}</ul>
// );

// export const figcaption = ({ children }: React.HTMLAttributes<HTMLElement>) => (
//   <figcaption>
//     <Text align="center" italic>
//       {children}
//     </Text>
//   </figcaption>
// );

// export const code = CodeBlock;

// export const dd = ({ children }: React.HTMLAttributes<HTMLElement>) => (
//   <dd className="ml-[2em]">
//     <Text>{children}</Text>
//   </dd>
// );

// export const dt = ({ children }: React.HTMLAttributes<HTMLElement>) => (
//   <dt>
//     <Heading as="h3" type="title2">
//       {children}
//     </Heading>
//   </dt>
// );

// export const inlineCode = InlineCode;

// function LinkForOrbitTextLink({ href, ...props }: { href: string }) {
//   return <Link to={href} {...props} />;
// }

// export const a = function Anchor({
//   children,
//   href,
//   orbitType,
// }: React.AnchorHTMLAttributes<HTMLAnchorElement> & {
//   orbitType?: React.ComponentProps<typeof TextLink>["type"];
// }) {
//   const isExternal = useIsUrlExternal(href);
//   const useExternalIcon = isExternal && typeof children === "string";
//   return (
//     <TextLink
//       asComponent={isExternal ? "a" : LinkForOrbitTextLink}
//       type={orbitType}
//       href={href}
//       external={isExternal}
//       iconRight={useExternalIcon && <NewWindow ariaLabel="Opens in new window" />}
//     >
//       {children}
//     </TextLink>
//   );
// };

// export function Callout({ icon = true, ...props }) {
//   return <Alert icon={icon} spaceAfter="large" {...props} />;
// }

// export default {
//   ...MDXComponents,
//   // Override default components
//   p,
//   h1,
//   h2,
//   h3,
//   h4,
//   h5,
//   h6,
//   hr,
//   ol,
//   ul,
//   blockquote,
//   table,
//   thead,
//   tbody,
//   tr,
//   td,
//   th,
//   figcaption,
//   code,
//   dd,
//   dt,
//   inlineCode,
//   a,
//   // Custom components available globally in MDX
//   Callout,
//   Palette,
//   IconList,
//   IllustrationList,
//   ElevationLevel,
//   ImagesDownloadList,
//   ComponentDesignTokens,
//   GlobalDesignTokens,
//   TailwindClassnames,
//   GithubContributors,
//   ComponentStatusList,
//   FigmaIframe,
//   InlineToken,
//   DocLayout,
// };
