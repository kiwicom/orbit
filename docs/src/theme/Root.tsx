import React from "react";
import { OrbitProvider, defaultTheme } from "@kiwicom/orbit-components";

// Default implementation, that you can customize
export default function Root({ children }: { children: React.ReactNode }) {
  return <OrbitProvider theme={defaultTheme}>{children}</OrbitProvider>;
}
