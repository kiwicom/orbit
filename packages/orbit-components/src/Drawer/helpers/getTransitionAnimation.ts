import { css, FlattenInterpolation, ThemeProps, DefaultTheme } from "styled-components";

import mq from "../../utils/mediaQuery";
import POSITIONS from "../consts";
import { Theme } from "../../defaultTheme";

const computedWidth = (width: string, isPrefixed: boolean) => `${isPrefixed ? "-" : ""}${width}`;

const transitionCss = ({ width, shown }: { width: string; shown?: boolean }) => (
  isPrefixed: boolean,
) => {
  return css`
    transform: translate3d(${shown ? "0, 0, 0" : `${computedWidth("100%", isPrefixed)} , 0, 0`});
    ${mq.largeMobile(css`
      transform: translate3d(${shown ? "0, 0, 0" : `${computedWidth(width, isPrefixed)} , 0, 0`});
    `)};
  `;
};

const getTransitionAnimation = ({
  width,
  shown,
  position,
  theme,
}: {
  width: string;
  shown?: boolean;
  position: "right" | "left";
  theme: Theme;
}): FlattenInterpolation<ThemeProps<DefaultTheme>> => {
  const resultCss = transitionCss({ width, shown });
  if ((position === POSITIONS.RIGHT && !theme.rtl) || (position === POSITIONS.LEFT && theme.rtl)) {
    return resultCss(false);
  }
  return resultCss(true);
};

export default getTransitionAnimation;
