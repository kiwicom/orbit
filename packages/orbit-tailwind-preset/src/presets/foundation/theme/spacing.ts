import foundation from "./defaultFoundation";

const spacings = {
  none: "0px",
  0: "0px",
  px: "1px",
  xxxs: foundation.space.xxxsmall,
  xxs: foundation.space.xxsmall,
  xs: foundation.space.xsmall,
  sm: foundation.space.small,
  md: foundation.space.medium,
  lg: foundation.space.large,
  xl: foundation.space.xlarge,
  xxl: foundation.space.xxlarge,
  xxxl: foundation.space.xxxlarge,
};

export default spacings;
