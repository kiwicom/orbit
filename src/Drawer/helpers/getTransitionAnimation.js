// @flow
import { css } from "styled-components";

import mq from "../../utils/mediaQuery";
import POSITIONS from "../consts";
import type { GetTransitionAnimation } from "./getTransitionAnimation";

const computedWidth = (width, isPrefixed) => `${isPrefixed ? "-" : ""}${width}`;

const transitionCss = ({ width, shown }) => isPrefixed => {
  return css`
    transform: translate3d(${shown ? "0, 0, 0" : `${computedWidth("100%", isPrefixed)} , 0, 0`});
    ${mq.largeMobile(css`
      transform: translate3d(${shown ? "0, 0, 0" : `${computedWidth(width, isPrefixed)} , 0, 0`});
    `)};
  `;
};

const getTransitionAnimation: GetTransitionAnimation = ({ width, shown, position, theme }) => {
  const resultCss = transitionCss({ width, shown });
  if ((position === POSITIONS.RIGHT && !theme.rtl) || (position === POSITIONS.LEFT && theme.rtl)) {
    return resultCss(false);
  }
  return resultCss(true);
};

export default getTransitionAnimation;
