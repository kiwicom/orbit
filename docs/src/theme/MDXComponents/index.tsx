import React from "react";
import MDXComponents from "@theme-original/MDXComponents";
import {
  Alert,
  Heading,
  Separator,
  Text,
  TableHead,
  TableBody,
  TableRow,
  TableCell,
} from "@kiwicom/orbit-components";

import BlockQuote from "../../components/MDXComponents/BlockQuote";
import { h1, h2, h3, h4, h5, h6 } from "../../components/MDXComponents/Heading";
import PropsTable from "../../components/MDXComponents/PropsTable";
import { a } from "../../components/MDXComponents/Anchor";
import * as Usage from "../../components/MDXComponents/Usage";
// // Import custom components to make them available globally
// import Palette from "../../ignore/Palette";
// import IconList from "../../ignore/IconList";
// import IllustrationList from "../../ignore/IllustrationList";
// import ElevationLevel from "../../ignore/ElevationLevel";
// import ImagesDownloadList from "../../ignore/ImagesDownloadList";
// import { ComponentDesignTokens, GlobalDesignTokens } from "../../ignore/DesignTokensList";
// import TailwindClassnames from "../../ignore/TailwindClassnames";
// import GithubContributors from "../../ignore/Github";
// import { ComponentStatusList } from "../../ignore/ComponentStatus";
import FigmaIframe from "../../components/MDXComponents/FigmaIframe";
import Guideline from "../../components/MDXComponents/Guideline";
import GuidelinesSideBySide, {
  Do,
  Dont,
} from "../../components/MDXComponents/GuidelinesSideBySide";
import ImageContainer from "../../components/MDXComponents/ImageContainer";
import FancyLink from "../../components/MDXComponents/FancyLink";
// import InlineToken from "../../ignore/InlineToken";
// import DocLayout from "../../ignore/DocLayout";

export const p = ({ children }: React.HTMLAttributes<HTMLParagraphElement>) => (
  <Text>{children}</Text>
);

export const hr = () => <Separator spaceAfter="largest" />;

export const ol = ({ children }: React.OlHTMLAttributes<HTMLOListElement>) => (
  <ol className="ml-400 [&_ol]:m-300 [&_ul]:m-300 list-outside list-decimal">{children}</ol>
);

export const blockquote = BlockQuote;

export const table = ({ children }: React.TableHTMLAttributes<HTMLTableElement>) => (
  <PropsTable>{children}</PropsTable>
);

export const thead = ({ children }: React.HTMLAttributes<HTMLTableSectionElement>) => (
  <TableHead>{children}</TableHead>
);

export const tbody = ({ children }: React.HTMLAttributes<HTMLTableSectionElement>) => (
  <TableBody>{children}</TableBody>
);

export const tr = ({ children }: React.HTMLAttributes<HTMLTableRowElement>) => (
  <TableRow>{children}</TableRow>
);

export const td = ({ children, align, valign }: React.TdHTMLAttributes<HTMLTableCellElement>) => (
  <TableCell
    as="td"
    align={align === "left" || align === "center" || align === "right" ? align : undefined}
    whiteSpace="normal"
    verticalAlign={valign}
  >
    {children}
  </TableCell>
);

export const th = ({ children }: React.ThHTMLAttributes<HTMLTableCellElement>) => (
  <TableCell as="th">{children}</TableCell>
);

export const ul = ({ children }: React.HTMLAttributes<HTMLUListElement>) => (
  <ul className="ml-400 [&_ol]:m-300 [&_ul]:m-300 list-outside list-disc">{children}</ul>
);

export const figcaption = ({ children }: React.HTMLAttributes<HTMLElement>) => (
  <figcaption>
    <Text align="center" italic>
      {children}
    </Text>
  </figcaption>
);

export const dd = ({ children }: React.HTMLAttributes<HTMLElement>) => (
  <dd className="ml-[2em]">
    <Text>{children}</Text>
  </dd>
);

export const dt = ({ children }: React.HTMLAttributes<HTMLElement>) => (
  <dt>
    <Heading as="h3" type="title2">
      {children}
    </Heading>
  </dt>
);

export function Callout({ icon = true, ...props }) {
  return <Alert icon={icon} spaceAfter="large" {...props} />;
}

export default {
  ...MDXComponents,
  // Override default components
  p,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  hr,
  ol,
  ul,
  blockquote,
  table,
  thead,
  tbody,
  tr,
  td,
  th,
  figcaption,
  dd,
  dt,
  a,
  // Custom components available globally in MDX
  FancyLink,
  Callout,
  ...Usage,
  // Palette,
  // IconList,
  // IllustrationList,
  // ElevationLevel,
  // ImagesDownloadList,
  // ComponentDesignTokens,
  // GlobalDesignTokens,
  // TailwindClassnames,
  // GithubContributors,
  // ComponentStatusList,
  FigmaIframe,
  Guideline,
  ImageContainer,
  GuidelinesSideBySide,
  Do,
  Dont,
  // InlineToken,
  // DocLayout,
};
