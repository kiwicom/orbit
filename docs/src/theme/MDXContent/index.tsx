import React, { type ReactNode } from "react";
import cx from "classnames";
import { MDXProvider } from "@mdx-js/react";
import MDXComponents from "@theme/MDXComponents";
import type { Props } from "@theme/MDXContent";

export default function MDXContent({ children }: Props): ReactNode {
  return (
    <div
      className={cx(
        "text-normal",
        "tb:rounded-400",
        "[&>*+*]:mt-400",
        "[&>*+h1]:mt-600 [&>*+h2]:mt-600 [&>*+h3]:mt-600 [&>*+h4]:mt-600 [&>*+h5]:mt-600",
        "[&>h1+*]:mt-300 [&>h2+*]:mt-300 [&>h3+*]:mt-300 [&>h4+*]:mt-300 [&>h5+*]:mt-300",
        "[&>p]:text-large",
        "[&_p]:leading-[1.5]",
        "[&_ul>li+li]:mt-300",
        "[&_ol>li+li]:mt-300",
        "[&>.orbit-docs-heading-anchor]:mt-600 [&>.orbit-docs-heading-anchor+*]:mt-0",
        "[&>h2:first-child]:mt-0",
        "[&>div+h2:nth-child(2)]:mt-0",
      )}
    >
      <MDXProvider components={MDXComponents}>{children}</MDXProvider>
    </div>
  );
}
