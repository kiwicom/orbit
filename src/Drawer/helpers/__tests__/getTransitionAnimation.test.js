// @flow
import getTransitionAnimation from "../getTransitionAnimation";
import theme from "../../../defaultTheme";
import POSITIONS from "../../consts";

/*
  It's very hard to match returned values from function, that returns deeply nested interpolation.
  Instead of some evaluation hack, snapshots are quiet easy for maintainability of this function.
 */

describe("getTransitionAnimation on the right", () => {
  const width = "320px";
  const position = POSITIONS.RIGHT;
  it("shown, should match snapshot", () => {
    expect(getTransitionAnimation({ width, shown: true, position, theme })).toMatchSnapshot();
  });
  it("not shown, should match snapshot", () => {
    expect(getTransitionAnimation({ width, shown: false, position, theme })).toMatchSnapshot();
  });
});

describe("getTransitionAnimation on the left", () => {
  const width = "320px";
  const position = POSITIONS.LEFT;
  it("shown, should match snapshot", () => {
    expect(getTransitionAnimation({ width, shown: true, position, theme })).toMatchSnapshot();
  });
  it("not shown, should match snapshot", () => {
    expect(getTransitionAnimation({ width, shown: false, position, theme })).toMatchSnapshot();
  });
});

describe("getTransitionAnimation on the right in RTL", () => {
  const width = "320px";
  const position = POSITIONS.RIGHT;
  it("shown, should match snapshot", () => {
    expect(
      getTransitionAnimation({ width, shown: true, position, theme: { ...theme, rtl: true } }),
    ).toMatchSnapshot();
  });
  it("not shown, should match snapshot", () => {
    expect(
      getTransitionAnimation({ width, shown: false, position, theme: { ...theme, rtl: true } }),
    ).toMatchSnapshot();
  });
});

describe("getTransitionAnimation on the left in RTL", () => {
  const width = "320px";
  const position = POSITIONS.LEFT;
  it("shown, should match snapshot", () => {
    expect(
      getTransitionAnimation({ width, shown: true, position, theme: { ...theme, rtl: true } }),
    ).toMatchSnapshot();
  });
  it("not shown, should match snapshot", () => {
    expect(
      getTransitionAnimation({ width, shown: false, position, theme: { ...theme, rtl: true } }),
    ).toMatchSnapshot();
  });
});
