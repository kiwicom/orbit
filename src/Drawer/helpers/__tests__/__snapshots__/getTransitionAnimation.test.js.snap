// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTransitionAnimation on the left in RTL not shown, should match snapshot 1`] = `
Array [
  "transform:translate3d(",
  "100% , 0, 0",
  ");",
  "@media ",
  [Function],
  "{",
  "transform:translate3d(",
  "320px , 0, 0",
  ");",
  ";}",
  ";",
]
`;

exports[`getTransitionAnimation on the left in RTL shown, should match snapshot 1`] = `
Array [
  "transform:translate3d(",
  "0, 0, 0",
  ");",
  "@media ",
  [Function],
  "{",
  "transform:translate3d(",
  "0, 0, 0",
  ");",
  ";}",
  ";",
]
`;

exports[`getTransitionAnimation on the left not shown, should match snapshot 1`] = `
Array [
  "transform:translate3d(",
  "-100% , 0, 0",
  ");",
  "@media ",
  [Function],
  "{",
  "transform:translate3d(",
  "-320px , 0, 0",
  ");",
  ";}",
  ";",
]
`;

exports[`getTransitionAnimation on the left shown, should match snapshot 1`] = `
Array [
  "transform:translate3d(",
  "0, 0, 0",
  ");",
  "@media ",
  [Function],
  "{",
  "transform:translate3d(",
  "0, 0, 0",
  ");",
  ";}",
  ";",
]
`;

exports[`getTransitionAnimation on the right in RTL not shown, should match snapshot 1`] = `
Array [
  "transform:translate3d(",
  "-100% , 0, 0",
  ");",
  "@media ",
  [Function],
  "{",
  "transform:translate3d(",
  "-320px , 0, 0",
  ");",
  ";}",
  ";",
]
`;

exports[`getTransitionAnimation on the right in RTL shown, should match snapshot 1`] = `
Array [
  "transform:translate3d(",
  "0, 0, 0",
  ");",
  "@media ",
  [Function],
  "{",
  "transform:translate3d(",
  "0, 0, 0",
  ");",
  ";}",
  ";",
]
`;

exports[`getTransitionAnimation on the right not shown, should match snapshot 1`] = `
Array [
  "transform:translate3d(",
  "100% , 0, 0",
  ");",
  "@media ",
  [Function],
  "{",
  "transform:translate3d(",
  "320px , 0, 0",
  ");",
  ";}",
  ";",
]
`;

exports[`getTransitionAnimation on the right shown, should match snapshot 1`] = `
Array [
  "transform:translate3d(",
  "0, 0, 0",
  ");",
  "@media ",
  [Function],
  "{",
  "transform:translate3d(",
  "0, 0, 0",
  ");",
  ";}",
  ";",
]
`;
